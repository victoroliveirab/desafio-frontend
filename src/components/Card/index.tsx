import Avatar from '@mui/material/Avatar';
import { useNavigate } from 'react-router-dom';
import styles from './styles.module.scss';

export interface ICard {
  data: {
    avatar: string;
    id: string;
    img: string;
    subtitle?: string;
    title: string;
    type: string;
    url: string;
  };
}

function Card({ data }: ICard) {
  const navigate = useNavigate();

  return (
    <div className={styles.container}>
      <div
        className={styles['thumb-wrapper']}
        onClick={() => {
          navigate(data.url, {
            state: data,
          });
        }}
      >
        <img src={data.img} alt={data.title} className={styles.thumb} />
      </div>
      <div className={styles.info}>
        <Avatar alt={data.type} src={data.avatar} />
        <div className={styles.info__text}>
          <h2>{data.title}</h2>
          <h4>{data.subtitle}</h4>
        </div>
      </div>
    </div>
  );
}

export default Card;
