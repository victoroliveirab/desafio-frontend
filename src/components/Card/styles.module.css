.container:not(:last-child) {
  width: 100%;
  height: auto;
  margin-bottom: 20px;
}

@media screen and (min-width: 576px) {
  .container {
    align-self: self-start;
  }
}

.thumb-wrapper {
  width: 100%;
  height: auto;
  margin-bottom: 10px;
}

.thumb {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

@media screen and (min-width: 576px) {
  .thumb-wrapper {
    height: 250px;
    object-fit: cover;
  }
}

.info {
  display: flex;
  width: 100%;
  line-height: 2;
}

.info > * + * {
  margin-left: 8px;
}

.info__text {
  height: 100%;
  min-height: 40px;
}

.info h2 {
  font-size: 1rem;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  text-overflow: ellipsis;
  -webkit-line-clamp: 2;
  overflow: hidden;
  margin: 0;
  max-height: 40px;
  line-height: 1.2;
  font-weight: normal;
}

.info h4 {
  font-size: 0.875rem;
  margin: 0;
  font-weight: normal;
}
