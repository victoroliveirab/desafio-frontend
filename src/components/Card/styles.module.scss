@import 'assets/sass/resources';

.container {
  width: 100%;
  height: auto;
  &:not(:last-child) {
    margin-bottom: 20px;
  }
  @include medium-screen {
    align-self: self-start;
  }
}

.thumb {
  @extend %fit-parent;
  object-fit: cover;

  &-wrapper {
    @extend %fit-parent-width;
    height: auto;
    margin-bottom: 10px;

    @include small-screen {
      height: 250px;
      object-fit: cover;
    }
  }
}

.info {
  @extend %fit-parent-width;
  display: flex;
  line-height: 2;

  &__text {
    @extend %fit-parent-height;
    min-height: 40px;
  }

  & h2 {
    font-size: 1rem;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    text-overflow: ellipsis;
    -webkit-line-clamp: 2;
    overflow: hidden;
    margin: 0;
    max-height: 40px;
    line-height: 1.2;
    font-weight: normal;
  }

  & h4 {
    font-size: 0.875rem;
    margin: 0;
    font-weight: normal;
  }

  & > * + * {
    margin-left: 8px;
  }
}
