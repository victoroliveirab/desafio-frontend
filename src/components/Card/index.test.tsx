import { MemoryRouter } from 'react-router-dom';
import { RenderResult, fireEvent, render } from 'test-utils';
import cardInfo from '__mocks__/card';
import Card from '.';

let component: RenderResult;

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('<Card />', () => {
  beforeEach(() => {
    component = render(
      <MemoryRouter>
        <Card data={cardInfo.data} />
      </MemoryRouter>
    );
  });
  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  test('should navigate when thumbnail clicked', () => {
    const thumbWrapper = component.getByTestId('thumb-wrapper');

    fireEvent.click(thumbWrapper);

    expect(mockNavigate).toBeCalledWith(cardInfo.data.url, {
      state: cardInfo.data,
    });
  });
});
