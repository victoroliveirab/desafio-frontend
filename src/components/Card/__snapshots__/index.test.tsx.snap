// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="thumb-wrapper"
          data-testid="thumb-wrapper"
        >
          <img
            alt="$1 VS $29,000 House!"
            class="thumb"
            src="https://i.ytimg.com/vi/1lIi1txDVzo/default.jpg"
          />
        </div>
        <div
          class="info"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          >
            <img
              alt="video"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://i.ytimg.com/vi/1lIi1txDVzo/default.jpg"
            />
          </div>
          <div
            class="info__text"
          >
            <h2>
              $1 VS $29,000 House!
            </h2>
            <h4>
              MrBeast Gaming • 7.9M views
            </h4>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="thumb-wrapper"
        data-testid="thumb-wrapper"
      >
        <img
          alt="$1 VS $29,000 House!"
          class="thumb"
          src="https://i.ytimg.com/vi/1lIi1txDVzo/default.jpg"
        />
      </div>
      <div
        class="info"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
        >
          <img
            alt="video"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://i.ytimg.com/vi/1lIi1txDVzo/default.jpg"
          />
        </div>
        <div
          class="info__text"
        >
          <h2>
            $1 VS $29,000 House!
          </h2>
          <h4>
            MrBeast Gaming • 7.9M views
          </h4>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
